org.eclipse.jdt.ui/ui/org/eclipse/refactoring/MakeStaticRefactoring.java

```diff
@@ -0,0 +1,885 @@
+		}
+	}
+
+	private boolean isOverriding(IType type) throws JavaModelException {	//TODO duplicate isOverridden()?
+		ITypeHierarchy hierarchy= type.newTypeHierarchy(null);
+		IType[] supertypes= hierarchy.getAllSupertypes(type);
+		for (IType supertype : supertypes) {
+			if (!(supertype.getElementName().equals("Object"))) { //$NON-NLS-1$
+				IMethod[] method= supertype.findMethods(fTargetMethod);
+				if (method != null) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	private void addInstanceAsParamIfUsed(RefactoringStatus status, AST ast, ASTRewrite rewrite, List<SingleVariableDeclaration> alreadyUsedParameters, String className, String paramName,
+			ITypeParameter[] classTypeParameters) throws JavaModelException {
+		SingleVariableDeclaration newParam= ast.newSingleVariableDeclaration();
+
+		//If generic TypeParameters exist in class the newParam type needs to be parameterized
+		if (classTypeParameters.length != 0) {
+			SimpleType simpleType= ast.newSimpleType(ast.newName(className));
+			ParameterizedType parameterizedType= ast.newParameterizedType(simpleType);
+			for (int i= 0; i < classTypeParameters.length; i++) {
+				SimpleType typeParameter= ast.newSimpleType(ast.newSimpleName(classTypeParameters[i].getElementName()));
+				parameterizedType.typeArguments().add(typeParameter);
+			}
+			newParam.setType(parameterizedType);
+
+		} else {
+			newParam.setType(ast.newSimpleType(ast.newName(className)));
+		}
+		newParam.setName(ast.newSimpleName(paramName));
+
+		//While refactoring, the method signature might change; ensure the revised method doesn't unintentionally override an existing one.
+		int parameterAmount= alreadyUsedParameters.size() + 1;
+		checkDuplicateMethod(status, parameterAmount);
+
+		//Add new parameter to method declaration arguments
+		ListRewrite lrw= rewrite.getListRewrite(fTargetMethodDeclaration, MethodDeclaration.PARAMETERS_PROPERTY);
+		lrw.insertFirst(newParam, null);
+
+		//Changes to fTargetMethodDeclaration's signature need to be adjusted in JavaDocs too
+		Javadoc javadoc= fTargetMethodDeclaration.getJavadoc();
+		if (javadoc != null) {
+			TagElement newParameterTag= ast.newTagElement();
+			newParameterTag.setTagName(TagElement.TAG_PARAM);
+			newParameterTag.fragments().add(ast.newSimpleName(paramName));
+			ListRewrite tagsRewrite= rewrite.getListRewrite(javadoc, Javadoc.TAGS_PROPERTY);
+			tagsRewrite.insertFirst(newParameterTag, null);
+		}
+	}
+
+	private void checkDuplicateMethod(RefactoringStatus status, int parameterAmount) throws JavaModelException {
+		String methodName= fTargetMethodDeclaration.getName().getIdentifier();
+		IMethodBinding methodBinding= fTargetMethodDeclaration.resolveBinding();
+		ITypeBinding typeBinding= methodBinding.getDeclaringClass();
+		IType type= (IType) typeBinding.getJavaElement();
+
+		IMethod method= org.eclipse.jdt.internal.corext.refactoring.Checks.findMethod(methodName, parameterAmount, false, type);
+
+		if (method != null) {
```

Comment:
In cases where such a check realizes a precondition for executing the method, better do the shortcut `if (method == null) { return; }`. It reduces the depth of the following code and makes it easier to follow.